### PR TITLE
Eric s geo coder

### PR DESCRIPTION
**Pull request by ESmith92**
refactor reverse geocoder and related classes

-refactor GoogleReverseGeocoder and spec
-GoogleReverseGeocoderParser and spec
-GoogleURLBuilder and spec

### CODE REVIEW

File: src/services/reverse-geocoder.php [hunk 1]
```diff
@@ -5,5 +5,6 @@
 use SearchApi\Models as Models;
 
 interface ReverseGeocoder {
+  function get_url( Models\GeoCoordinate $coords );
```
Comment (rraub) [inline src/services/reverse-geocoder.php:8]: Is anything actually calling get_url? even inside of the `GoogleReverseGeocoder` it doesn't call it. I get the feeling that the ReverseGeocoder interface doesn't need to require a function get_url()

Comment (ESmith92) [inline src/services/reverse-geocoder.php:8]: No it really doesn't need the get _url ().. just wanted to be able to test the returned url in the geo-encoder

**Comment by rraub** [comment]
you have some merge conflicts that you'll need to resolve before we can merge this in

**Comment by kenprice** [comment]
@ESmith92 
To resolve merge conflict, do the following:
1. `git checkout develop` to check out your local `develop` branch.
2. `git pull origin develop` to pull latest `develop` branch from GitHub.
3. `git checkout EricS-GeoCoder` to check out your branch.
4. `git merge develop` to merge `develop` into your `EricS-GeoCoder` branch. Oh no! Merge conflict! No worries.
5. Note the merge conflict message from the last step. If you do `git status`, you'll see a bunch of files in green, followed by this:
    `deleted by us:   src/providers/reverse-geocoder-class.php`
    `deleted by us:   test/spec/SearchApi/Providers/ReverseGeocoderClassSpec.php`
6. For each of these, do a `git rm <FULL FILE PATH HERE>`. So `git rm src/providers/reverse-geocoder-class.php` for the first file. Do the same with the second. You need to do this since `develop` has these files, but in your branch they have been deleted. When the merge occurred, git isn't sure whether these files should still exist or not, hence the merge conflict. `git rm ...` will tell git that you'd like these files deleted.
7. `git commit`, use the same git commit message when prompted. A merge is basically a giant commit that takes the changes you don't have and apply them as one big commit. When a merge conflict had occurred, the commit was suspended. That's why you'd need to do another `git commit`. A `git commit` without a message will put git in some interactive mode that asks for the message. If you have followed these steps correctly, you should see something like `Merge X to Y` as the commit message.
8. Conflict should now be resolved. `git push origin EricS-GeoCoder` and cross your fingers.

**Pull request merged**